### PR TITLE
Pallet loans ref fully benchmarked

### LINKED ISSUE #1249: Pallet loans refactor: benchmarks
### Which part of the code is the issue addressing?

- Feature

### Description

Perform analogous benchmarks into `pallet-loans-ref`.
- Essentials: #1242
- Remaining: #1260

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

`pallet-loans-ref` fully benchmarked.

Fixes #1249

## Changes and Descriptions

- Add benchmarks for the rest of `pallet-loans-ref` methods.
- No coupled dependencies for benchmarks.
- Removed `MaxRates` benchmark parameter because it's almost negligible. 100 rates equivalent to 1 loan of overhead. Now it takes the maximum value in all benchmarks.

### CODE REVIEW

File: pallets/loans-ref/src/types.rs [hunk 1]
```diff
@@ -767,4 +767,10 @@ mod test_utils {
 			self
 		}
 	}
+
+	impl<T: Config> ActiveLoan<T> {
+		pub fn set_maturity(&mut self, duration: Duration) {
+			self.info.schedule.maturity = Maturity::Fixed(duration.as_secs());
+		}
+	}
```
Comment (mustermeiszer) [inline pallets/loans-ref/src/types.rs:775]: Should this maybe also be guarded by benchmark feature flag?

**Review by mustermeiszer — APPROVED** [review]
Just one little comment. Nicely light-coupled benchmarks ❤️

File: pallets/loans-ref/src/types.rs (hunk 1 above)
Comment (lemunozm) [reply pallets/loans-ref/src/types.rs:775]: Yes, it already is

**Comment by lemunozm** [comment]
Thanks for your fast review 🚀

**Pull request merged**